compiler/src/dotty/tools/dotc/transform/SuperAccessors.scala

```diff
@@ -153,7 +153,9 @@ class SuperAccessors(thisPhase: DenotTransformer) {
         }
     }
 
-    val needAccessor = name.isTermName && (
+    val needAccessor =
+      name.isTermName                // Types don't need super accessors
+      && !sym.isInlineMethod && (    // Inline methods can't be overridden, don't need superaccessors
```

Comment:
Based on the comment, shouldn't we apply this restriction to all final symbols?

Reply:
I believe that the comment is wrong. The reason for not generating this accessor is that there won't be a call at runtime to this method.